custom/icds/messaging/indicators.py

```diff
@@ -224,3 +274,72 @@ def convert_to_date(date):
             messages.append(self.render_template(context, language_code=language_code))
 
         return messages
+
+
+class LSAggregatePerformanceIndicator(LSIndicator):
+    template = 'ls_aggregate_performance.txt'
+
+    @property
+    @memoized
+    def restore_user(self):
+        return OTARestoreCommCareUser(self.domain, self.user)
+
+    @property
+    @memoized
+    def report_configs(self):
+        report_ids = [
+            HOME_VISIT_REPORT_ID,
+            THR_REPORT_ID,
+            CHILDREN_WEIGHED_REPORT_ID,
+            DAYS_AWC_OPEN_REPORT_ID,
+        ]
+        app = get_app(self.domain, SUPERVISOR_APP_ID)
+        return {
+            report_config.report_id: report_config
+            for module in app.modules if isinstance(module, ReportModule)
+            for report_config in module.report_configs
+            if report_config.report_id in report_ids
+        }
+
+    def get_report_fixture(self, report_id):
+        return ReportFixturesProvider.report_config_to_fixture(
+            self.report_configs[report_id], self.restore_user
+        )
+
+    @property
+    def visits_fixture(self):
+        return self.get_report_fixture(HOME_VISIT_REPORT_ID)
+
+    @property
+    def thr_fixture(self):
+        return self.get_report_fixture(THR_REPORT_ID)
+
+    @property
+    def weighed_fixture(self):
+        return self.get_report_fixture(CHILDREN_WEIGHED_REPORT_ID)
+
+    @property
+    def days_open_fixture(self):
+        return self.get_report_fixture(DAYS_AWC_OPEN_REPORT_ID)
+
+    def get_value_from_fixture(self, fixture, attribute):
+        xpath = './rows/row[@is_total_row="True"]/column[@id="{}"]'.format(attribute)
+        return fixture.findall(xpath)[0].text
+
+    def get_messages(self, language_code=None):
+        visit_on_time = self.get_value_from_fixture(self.visits_fixture, 'visit_on_time')
+        visits = self.get_value_from_fixture(self.visits_fixture, 'count')
+        thr_gte_21 = self.get_value_from_fixture(self.thr_fixture, 'open_ccs_thr_gte_21')
+        thr_count = self.get_value_from_fixture(self.thr_fixture, 'open_count')
+        num_weigh = self.get_value_from_fixture(self.weighed_fixture, 'open_weighed')
+        num_weigh_avail = self.get_value_from_fixture(self.weighed_fixture, 'open_count')
+        num_days_open = self.get_value_from_fixture(self.days_open_fixture, 'awc_opened_count')
+
+        context = {
+            "visits": "{} / {}".format(visit_on_time, visits),
+            "thr_distribution": "{} / {}".format(thr_gte_21, thr_count),
+            "children_weighed": "{} / {}".format(num_weigh, num_weigh_avail),
+            "days_open": "{}".format(int(num_days_open) // len(self.awc_locations)),
```

Comment:
also they wanted this rounded to the nearest whole number so i think it would have to be:

`int(round(1.0 * x / y))`